test/Microsoft.TemplateEngine.TestHelper/PackageManager.cs

```diff
@@ -45,11 +48,24 @@ public string GetNuGetPackage(string templatePackName)
                     return packagePath;
                 }
 
-                Task<string> downloadTask = DownloadPackageAsync(templatePackName);
-                downloadTask.Wait();
-                string downloadedPackage = downloadTask.Result;
-                _installedPackages[templatePackName] = downloadedPackage;
-                return downloadedPackage;
+                for (int retry = 0; retry < 5; retry++)
+                {
+                    try
+                    {
+                        Task<string> downloadTask = DownloadPackageAsync(templatePackName, log: log);
+                        downloadTask.Wait();
```

Comment:
It would be nice if instead of `Wait`ing on the tasks the method was async and we used `await`.